repo/locking_storage.go

```diff
@@ -0,0 +1,22 @@
+package repo
+
+import (
+	"github.com/kopia/kopia/internal/epoch"
+	"github.com/kopia/kopia/repo/content"
+	"github.com/kopia/kopia/repo/content/indexblob"
+	"github.com/kopia/kopia/repo/format"
+)
+
+// GetLockingStoragePrefixes Return all prefixes that may be maintained by Object Locking.
+func GetLockingStoragePrefixes() []string {
+	var prefixes []string
+	// collect prefixes that need to be locked on put
+	for _, prefix := range content.PackBlobIDPrefixes {
+		prefixes = append(prefixes, string(prefix))
+	}
+
+	prefixes = append(prefixes, indexblob.V0IndexBlobPrefix, epoch.EpochManagerIndexUberPrefix, format.KopiaRepositoryBlobID,
+		format.KopiaBlobCfgBlobID)
+
+	return prefixes
+}
```

Comment:
I'm not sure how to get a list of all valid prefixes to build an exclude list.  I'm also concerned that it would cause user breakage.  I am not sure that a new blob would be properly tested with extension, resulting in creating a retained blob that must be removed (per kopia policy) but cannot  (due to retention rules) and not catching this until after the code has been deployed to production.
I'm open to ideas though.

Reply:
I think this is a task for another day; I recognize this change here is just a decomposition of existing code for reuse. I think it would be safer, but we can revisit it later.